server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1667,7 +1666,6 @@ public DeleteResult delete(Delete delete) throws IOException {
                 localCheckpointTracker.markSeqNoAsPersisted(deleteResult.getSeqNo());
             }
             deleteResult.setTook(System.nanoTime() - delete.startTime());
```

Comment:
`Math.max(0, took)` in the setter? ;)

Reply:
Actually, even if the start time is coming out of a cached value it will lag behind, ever be ahead of the current nanoTime so I think this is fine right?